pkg/codegen/go/gen.go

```diff
@@ -676,13 +699,12 @@ func (pkg *pkgContext) contextForExternalReference(t schema.Type) *pkgContext {
 		}
 	}
 
-	extPkgCtx := &pkgContext{
-		pkg:              extPkg,
-		importBasePath:   goInfo.ImportBasePath,
-		pkgImportAliases: pkgImportAliases,
-		modToPkg:         goInfo.ModuleToPackage,
-	}
-	return extPkgCtx
+	maps := generatePackageContextMap(pkg.tool, extPkg, goInfo)
```

Comment:
I'm concerned that this will have performance implications. I'll cache the result, but unfortunately global analysis is necessary for correctness.

Reply:
Yeah it sounds like it should be not that hard to cache the result at least inside `pkgInfo`, from skimming I didn't see how exactly you did it, but caching the global anaylsis is good. 

You can try to regen the azure-native with these changes and se how much longer it took (actually downstream checks may have done that for us) so that's a data point on perf.

Expecting it to be quite acceptable.